bigtable-client-core-parent/bigtable-hbase/src/main/java/com/google/cloud/bigtable/hbase/wrappers/veneer/BigtableHBaseVeneerSettings.java

```diff
@@ -759,74 +738,63 @@ private void configureRetryableCallSettings(
 
   private void configureNonRetryableCallSettings(
       UnaryCallSettings.Builder<?, ?> unaryCallSettings, OperationTimeouts operationTimeouts) {
-    unaryCallSettings.setRetryableCodes(Collections.<StatusCode.Code>emptySet());
+    unaryCallSettings.setRetryableCodes(Collections.emptySet());
 
-    // NOTE: attempt timeouts are not configured for non-retriable rpcs
     if (operationTimeouts.getOperationTimeout().isPresent()) {
       unaryCallSettings
           .retrySettings()
           .setLogicalTimeout(operationTimeouts.getOperationTimeout().get());
     }
   }
 
+  /**
+   * Build a source of truth for timeout related settings.
+   *
+   * <p>This will combine defaults with end user provided overrides to configure response, attempt
+   * and operation timeouts. There 3 types of operations that are configured here:
+   *
+   * <ul>
+   *   <li>Unary - ReadRow, MutateRow, CheckAndMutate, ReadModifyWrite
+   *   <li>Scans - ReadRows (and BulkReadRow)
+   *   <li>Bulk Mutations
+   * </ul>
+   */
   private ClientOperationTimeouts buildCallSettings() {
-    boolean useTimeouts = configuration.getBoolean(BIGTABLE_USE_TIMEOUTS_KEY, true);
-
-    Optional<Duration> DEFAULT_BATCH_BULK_MUTATE_OVERALL_TIMEOUT =
-        Optional.of(Duration.ofMinutes(20));
-    // Set 20 minutes default timeout for batch jobs.
-    // The default override the deprecated BIGTABLE_LONG_RPC_TIMEOUT_MS_KEY.
-    Optional<Duration> bulkMutateOverallTimeout =
-        configuration.getBoolean(BIGTABLE_USE_BATCH, false)
-            ? extractOverallTimeout(BIGTABLE_MUTATE_RPC_TIMEOUT_MS_KEY)
-                .or(DEFAULT_BATCH_BULK_MUTATE_OVERALL_TIMEOUT)
-            : extractOverallTimeout(
-                BIGTABLE_MUTATE_RPC_TIMEOUT_MS_KEY,
-                BigtableOptionsFactory.BIGTABLE_LONG_RPC_TIMEOUT_MS_KEY);
+    Optional<Duration> defaultBatchMutateOverallTimeout =
+        DEFAULT_TIMEOUTS.bulkMutateTimeouts.operationTimeout;
+    if (configuration.getBoolean(BIGTABLE_USE_BATCH, false)) {
+      defaultBatchMutateOverallTimeout = Optional.of(Duration.ofMinutes(20));
```

Comment:
why is it 20 miuntes?

Reply:
just a carryover from the original. I was toying with the idea of dropping

Reply:
Lets leave it for now